src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseTestCaseService.java

```diff
@@ -125,41 +135,132 @@ public boolean generateTestCasesFromFeedbacks(List<Feedback> feedbacks, Programm
     }
 
     /**
-     * Updates an incoming result with the information of the exercises test cases. This update includes: - Checking which test cases were not executed as this is not part of the
-     * bamboo build (not all test cases are executed in an exercise with sequential test runs) - Recalculating the score based based on the successful test cases weight vs the
-     * total weight of all test cases.
+     * Updates an incoming result with the information of the exercises test cases. This update includes:
+     * - Checking which test cases were not executed as this is not part of the bamboo build (not all test cases are executed in an exercise with sequential test runs)
+     * - Checking the due date and the afterDueDate flag
+     * - Recalculating the score based based on the successful test cases weight vs the total weight of all test cases.
      *
      * If there are no test cases stored in the database for the given exercise (i.e. we have a legacy exercise) or the weight has not been changed, then the result will not change
      *
      * @param result   to modify with new score, result string & added feedbacks (not executed tests)
      * @param exercise the result belongs to.
-     * @return the updated result
+     * @param isStudentParticipation boolean flag indicating weather the participation of the result is not a solution/template participation.
+     * @return Result with updated feedbacks, score and result string.
      */
-    public Result updateResultFromTestCases(Result result, ProgrammingExercise exercise) {
+    @Transactional
+    public Result updateResultFromTestCases(Result result, ProgrammingExercise exercise, boolean isStudentParticipation) {
+        boolean removeAfterDueDateTests = isStudentParticipation && exercise.getDueDate() != null && ZonedDateTime.now().isBefore(exercise.getDueDate());
```

Comment:
removeAfterDueDateTests sounds like a strange variable name

Reply:
You are right, I made it more explicit: `shouldTestsWithAfterDueDateFlagBeRemoved`